dex/msgjson/types.go

```diff
@@ -868,7 +871,14 @@ type ConnectResult struct {
 	Sig                 Bytes          `json:"sig"`
 	ActiveOrderStatuses []*OrderStatus `json:"activeorderstatuses"`
 	ActiveMatches       []*Match       `json:"activematches"`
-	Score               int32          `json:"score"`
+	Reputation          Reputation     `json:"reputation"`
+}
+
+// Reputation is the user's reputation breakdown.
+type Reputation struct {
+	Score     int     `json:"score"`
+	RawScore  int32   `json:"rawScore"`
+	Privilege float64 `json:"privilege"`
```

Comment:
I'm still on board with the gist of these changes, especially in a tiered framework introduced in #1120.

One thing to note about #1120 is that I've merged the cancel rate outcome tracking into the score computation instead of having it as a separate banable item.  I've also made a TODO in the Connect result with a violations slice, but it can also work like this with `Reputation`, perhaps with details of the outcomes that impacted the user's score.

I feel like the way forward it to first merge the hd keys PR (#1015), then fidelity bonds with tiered accounts (#1120 ), then this to complete it.

Reply:
> I've merged the cancel rate outcome tracking into the score computation

Love it.